lib/client.js

```diff
@@ -161,9 +162,11 @@ var GiantSwarm = (function () {
       if (clusterId) {
         headers[CLUSTER_ID_HEADER] = clusterId;
       }
-      request
-        .get(apiEndpoint + uri)
-        .set(headers)
+      var r = request.get(apiEndpoint + uri);
+      if (params !== null) {
```

Comment:
you can make your code a lot easier here: `request.get(..).query(params || "").set(headers).end...`